charts/tidb-cluster/values.yaml

```diff
@@ -417,39 +421,27 @@ binlog:
     # refer to https://kubernetes.io/docs/concepts/storage/storage-classes
     storageClassName: local-storage
     storage: 10Gi
-    # the number of the concurrency of the downstream for synchronization. The bigger the value,
-    # the better throughput performance of the concurrency (16 by default)
-    workerCount: 16
-    # the interval time (in seconds) of detect pumps' status (default 10)
-    detectInterval: 10
     # disbale detect causality
     disableDetect: false
-    # disable dispatching sqls that in one same binlog; if set true, work-count and txn-batch would be useless
-    disableDispatch: false
-    # # disable sync these schema
-    ignoreSchemas: "INFORMATION_SCHEMA,PERFORMANCE_SCHEMA,mysql,test"
     # if drainer donesn't have checkpoint, use initial commitTS to initial checkpoint
     initialCommitTs: 0
-    # enable safe mode to make syncer reentrant
-    safeMode: false
-    # the number of SQL statements of a transaction that are output to the downstream database (20 by default)
-    txnBatch: 20
-    # downstream storage, equal to --dest-db-type
-    # valid values are "mysql", "pb", "kafka"
-    destDBType: pb
-    mysql: {}
-      # host: "127.0.0.1"
-      # user: "root"
-      # password: ""
-      # port: 3306
-      # # Time and size limits for flash batch write
-      # timeLimit: "30s"
-      # sizeLimit: "100000"
-    kafka: {}
-      # only need config one of zookeeper-addrs and kafka-addrs, will get kafka address if zookeeper-addrs is configed.
-      # zookeeperAddrs: "127.0.0.1:2181"
-      # kafkaAddrs: "127.0.0.1:9092"
-      # kafkaVersion: "0.8.2.0"
+    # Please refer to https://github.com/pingcap/tidb-binlog/blob/master/cmd/drainer/drainer.toml for the default
+    # drainer configurations (change to the tags of your drainer version),
+    # just follow the format in the file and configure in the 'config' section
+    # as below if you want to customize any configuration.
+    config: |
```

Comment:
This causes the old configuration key invalid, which is a in-compatible change

Reply:
yes, this is an in-compatible change like: https://github.com/pingcap/tidb-operator/pull/638, i think we should do it before GA. @tennix

Reply:
What about we keep these two ways, but make it clear in the document that we do not maintain the config for the original way and users should migrate to the new way if they want to configure any new parameters that are not exposed in values.yaml.

Reply:
Have we decided to make this change incompatible?

Reply:
What do you think? @tennix

Reply:
Agree with @DanielZhangQD we should not make this incompatible change.

Reply:
I think we can keep both for binlog configurations to keep compatible.

Reply:
comments addressed.